app/src/interfaces/map/map.vue

```diff
@@ -55,153 +83,234 @@ export default defineComponent({
-						? mapbox_sources[background]
-						: {
-								version: 8,
-								glyphs:
-									'https://basemaps.arcgis.com/arcgis/rest/services/OpenStreetMap_v2/VectorTileServer/resources/fonts/{fontstack}/{range}.pbf',
-								sprite: 'https://rawgit.com/lukasmartinelli/osm-liberty/gh-pages/sprites/osm-liberty',
-								layers: [
-									{
-										id: background,
-										source: background,
-										type: 'raster',
-									},
-								],
-								sources,
-						  },
+				container: container.value!,
+				style: style.value,
 				center: [props.longitude, props.latitude],
 				zoom: props.zoom,
 				attributionControl: false,
 			});
 
+			const fitDataControl = new ControlButton('mapboxgl-ctrl-fitdata', fitDataBounds);
+			const snakeGeometryType = snakeCase(props.geometryType);
+			const draw = new MapboxDraw({
+				displayControlsDefault: false,
+				controls: {
+					trash: true,
+					[snakeGeometryType]: true,
+				},
+				defaultMode: `draw_${snakeGeometryType}`,
+				styles: drawStyle,
+			});
 			map.addControl(new maplibre.NavigationControl(), 'top-left');
 			map.addControl(new maplibre.GeolocateControl(), 'top-left');
-			map.addControl(locateMarkerControl, 'top-left');
-			map.addControl(addMarkerControl.value, 'top-left');
-
-			renderMarker(props.value, null);
-		}
-
-		function onDragEnd() {
-			const lngLat = marker.value?.getLngLat();
-
-			if (lngLat) saveLocation(lngLat);
-		}
-
-		function saveLocation(lngLat: LngLat) {
-			const coordinates = proj4('EPSG:4326', props.projection, [lngLat.lng, lngLat.lat]);
-
-			if (props.type === 'json') {
-				emit('input', {
-					type: 'Point',
-					coordinates,
+			map.addControl(new ControlGroup(fitDataControl), 'top-left');
+			map.addControl(draw as IControl, 'top-left');
+
+			map.on('load', async () => {
+				await addMarkerImage().catch(() => {});
+				map.on('draw.create', handleDrawUpdate);
+				map.on('draw.delete', handleDrawUpdate);
+				map.on('draw.update', handleDrawUpdate);
+				map.once('resize', () => {
+					if (props.loading) {
+						watch(() => props.loading, addInitialValue);
+					} else {
+						addInitialValue();
+					}
 				});
-			}
-			if (props.type === 'csv') {
-				emit('input', coordinates);
-			}
-		}
-
-		function toggleMarker() {
-			if (props.value !== null) return emit('input', null);
-			saveLocation(map.getCenter());
-		}
-
-		function locateMarker(animate = true) {
-			const lngLat = marker.value === null ? { lng: props.longitude, lat: props.latitude } : marker.value.getLngLat();
-
-			map.flyTo({
-				center: lngLat,
-				animate,
-				speed: 1.2,
+				map.resize();
 			});
-		}
 
-		function renderMarker(newVal: ModelValue, oldVal: ModelValue) {
-			if (newVal === null) {
-				marker.value?.remove();
-				marker.value = null;
-				addMarkerControl.value?.activate(false);
-				return;
+			return () => {
+				map.remove();
+			};
+
+			function addMarkerImage() {
+				return new Promise((resolve, reject) => {
+					map.loadImage(
```

Comment:
Can this be done with a local data string svg? Would be nice to dynamically use `var(--primary)` for the marker

Reply:
An SVG  loaded as an image in `maplibre-gl` would lose access to css variables. Also, you can't directly add SVG data, you have to parse it into an image first (here's the hack: https://github.com/mapbox/mapbox-gl-js/issues/5529#issuecomment-340011876).

Reply:
We can dynamically read the actual value of the css var the moment we're initializing this

> Also, you can't directly add SVG data, you have to parse it into an image first

meh that's a shame